src/Features/CSharp/Portable/CodeRefactorings/ConvertToInterpolatedString/ConvertToInterpolatedStringRefactoringProvider.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Simplification;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ConvertToInterpolatedString
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.ConvertToInterpolatedString), Shared]
+    internal partial class ConvertToInterpolatedStringRefactoringProvider :
+        AbstractConvertToInterpolatedStringRefactoringProvider
+            <InterpolatedStringExpressionSyntax,
+             InterpolatedStringContentSyntax,
+             InterpolationSyntax,
+             InvocationExpressionSyntax,
+             ExpressionSyntax,
+             ArgumentSyntax,
+             LiteralExpressionSyntax>
```

Comment:
how should I format this? Also, is there a record for most generic type arguments?